controller/src/main/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasks.java

```diff
@@ -180,10 +172,18 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
                             DeleteTableEvent deleteEvent = new DeleteTableEvent(scope, kvtName, requestId, UUID.fromString(id));
                             return eventHelper.addIndexAndSubmitTask(deleteEvent,
                                     () -> kvtMetadataStore.setState(scope, kvtName, KVTableState.DELETING, context, executor))
-                                    .thenCompose(x -> eventHelper.checkDone(() -> isDeleted(scope, kvtName, context)))
```

Comment:
why not pass the operation context? that will avoid fetching the table names etc again from segment store

Reply:
When context is not present it fetches from cache.